crates/test-utils/src/deploy_helpers.rs

```diff
@@ -831,6 +746,71 @@ impl KakarotTestEnvironmentContext {
         Self { sequencer, kakarot_client, kakarot, kakarot_contract, evm_contracts }
     }
 
+    /// Initializes the Kakarot test environment and deploys the Kakarot system and EVM contracts.
+    async fn initialize_without_dumped_state(
+        sequencer: TestSequencer,
+        starknet_provider: Arc<JsonRpcClient<HttpTransport>>,
+    ) -> Self {
+        let starknet_account = sequencer.account();
+
+        // Define the expected funded amount for the Kakarot system
+        let expected_funded_amount = FieldElement::from_dec_str("1000000000000000000").unwrap();
+
+        // Deploy the Kakarot system
+        let kakarot = deploy_kakarot_system(&sequencer, EOA_WALLET.clone(), expected_funded_amount).await;
+
+        let starknet_deployer_account = deploy_deployer_account(starknet_provider.clone(), &starknet_account).await;
+
+        // Create a Kakarot client
+        let kakarot_client = KakarotClient::new(
+            StarknetClientConfig::new(
+                Network::JsonRpcProvider(sequencer.url()),
+                kakarot.kakarot_address,
+                kakarot.proxy_class_hash,
+            ),
+            starknet_provider,
```

Comment:
why do we give a `starknet_provider` here while we also seem to give it 3 lines above at `Network::JsonRpcProvider(sequencer.url()),`

Reply:
`KakarotClient` needs a provider 

https://github.com/kkrt-labs/kakarot-rpc/blob/0f4a260c8e8e77f637b0930052015f34c87ce533/crates/core/src/client/mod.rs#L64-L69

while `StarknetConfig` needs a `Network` which is an enum and we are using the `JsonRpcProvider`, passing the url of the sequencer

Reply:
this is the current reason following the line of depth and params of struct and constructor. But doesn't this seem odd to you?

Reply:
So following this [pr](https://github.com/kkrt-labs/kakarot-rpc/pull/301), we had to add it for simulate transaction because:
- we couldn't simulate tx on some networks and had to add this network in the client to do a match
- Provider doesn't have a `url` and it would have been complicated to add a wrapper that would have a trait that does this

Perhaps we can check if it's possible to do simulate tx on other networks now and so we could remove this. I'll create an issue

Reply:
Issue here https://github.com/kkrt-labs/kakarot-rpc/issues/570